test/components/patch_serviceaccount_test.go

```diff
@@ -0,0 +1,131 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	retryInterval  = time.Second * 5
+	timeout        = time.Minute * 5
+	contextTimeout = 10
+)
+
+type componentTestCase struct {
+	namespace string
+	platforms []testutil.Platform
+}
+
+func TestPatchServiceAccount(t *testing.T) {
+	client := testutil.CreateKubeClient(t)
+	componentTestCases := []componentTestCase{
+		{
+			namespace: "cert-manager",
+			platforms: []testutil.Platform{testutil.PlatformPacket, testutil.PlatformPacketARM},
+		},
+		{
+			namespace: "projectcontour",
+			platforms: []testutil.Platform{testutil.PlatformPacket, testutil.PlatformPacketARM},
+		},
+		{
+			namespace: "dex",
+			platforms: []testutil.Platform{testutil.PlatformPacket, testutil.PlatformPacketARM},
+		},
+		{
+			namespace: "external-dns",
+			platforms: []testutil.Platform{testutil.PlatformPacket, testutil.PlatformPacketARM},
+		},
+		{
+			namespace: "reboot-coordinator",
+			platforms: []testutil.Platform{testutil.PlatformPacket, testutil.PlatformPacketARM},
+		},
+		{
+			namespace: "gangway",
+			platforms: []testutil.Platform{testutil.PlatformPacket, testutil.PlatformPacketARM},
+		},
+		{
+			namespace: "httpbin",
+			platforms: []testutil.Platform{testutil.PlatformPacket, testutil.PlatformPacketARM},
+		},
+		{
+			namespace: "metallb-system",
+			platforms: []testutil.Platform{testutil.PlatformPacket, testutil.PlatformPacketARM},
+		},
+		{
+			namespace: "openebs",
+			platforms: []testutil.Platform{testutil.PlatformPacket, testutil.PlatformPacketARM},
+		},
+		{
+			namespace: "monitoring",
+			platforms: []testutil.Platform{testutil.PlatformPacket, testutil.PlatformPacketARM},
+		},
+		{
+			namespace: "rook",
+			platforms: []testutil.Platform{testutil.PlatformPacket, testutil.PlatformPacketARM},
+		},
+	}
+
+	for _, tc := range componentTestCases {
+		tc := tc
+		t.Run(tc.namespace, func(t *testing.T) {
+			t.Parallel()
+
+			if !testutil.IsPlatformSupported(t, tc.platforms) {
+				t.Skip()
+			}
+
+			if err := wait.PollImmediate(
+				retryInterval, timeout, checkDefaultServiceAccountPatch(client, tc),
+			); err != nil {
+				t.Fatalf("%v", err)
+			}
+		})
+	}
+}
+
+func checkDefaultServiceAccountPatch(client kubernetes.Interface, tc componentTestCase) wait.ConditionFunc {
+	return func() (done bool, err error) {
+		ctx, cancel := context.WithTimeout(context.Background(), contextTimeout*time.Second)
+		defer cancel()
+
+		sa, err := client.CoreV1().ServiceAccounts(tc.namespace).Get(ctx, "default", metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("error getting service account: %v", err)
+		}
+
+		automountServiceAccountToken := *sa.AutomountServiceAccountToken
+
+		if automountServiceAccountToken != false {
+			//nolint:lll
+			return false, fmt.Errorf("Service account for %v namespace not patched. Expected %v got %v", tc.namespace, false, automountServiceAccountToken)
```

Comment:
Fixed casing, shuffled a variable around and used `%q` so it's quoted.
```suggestion
			return false, fmt.Errorf("service account for namespace %q was not patched. Expected %v got %v", tc.namespace, false, automountServiceAccountToken)
```